### DISCUSSION TITLE
Security tutorial should salt hashes

### DISCUSSION THREAD

**Discussion by corneliusroemer** [fastapi/fastapi#10683] [Questions] [answered]
I worked through the security tutorial.

An important omission is a lack of per-user cryptographic salt. Otherwise rainbow tables of common passwords can be used to crack the key.

A per user salt should be added to the tutorial to teach best practices.

https://en.m.wikipedia.org/wiki/Salt_(cryptography)#:~:text=In%20cryptography%2C%20a%20salt%20is,needed%20for%20a%20successful%20attack.

**Comment by nymous** [comment, marked as answer]
Hello,
The tutorial uses `passlib`, which automatically takes care of generating a random salt for you: https://passlib.readthedocs.io/en/stable/narr/hash-tutorial.html#hashing (and https://passlib.readthedocs.io/en/stable/narr/context-tutorial.html#basic-usage for an example using CryptContext specifically)
